### PR TITLE
feat(language-service): allow retreiving synchronized analyzed NgModules

### PR DESCRIPTION
**Pull request by ayazhafiz**
Sometimes modules retreived from the language service need to be
synchronized to the last time they were updated, and not updated
on-the-fly. This PR adds a flag to
`TypeScriptServiceHost#getAnalyzedModules` that retreives cached
analyzed NgModules rather than potentially recomputing them.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Feature

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/language-service/src/typescript_host.ts [hunk 1]
```diff
@@ -152,9 +152,12 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
    * and templateReferences.
    * In addition to returning information about NgModules, this method plays the
    * same role as 'synchronizeHostData' in tsserver.
+   * @param ensureSynchronized if true, returns the set of analyzedModules that is already cached.
```
Comment (kyliau) [inline packages/language-service/src/typescript_host.ts]: Sorry, this seems contrary to the idea of `ensureSynchronized`.
The flag means that if true, "do the work to make sure the program is synchronized".
The cached result should be returned if `ensureSynchronized` is false, i.e. no need to make sure it is up-to-date.

tldr; this should default to true.
See https://github.com/microsoft/TypeScript/blob/e60d28e6f76e7bbb25e06a1742dc9a24fd7a1f16/src/server/project.ts#L509

**Review by kyliau — CHANGES_REQUESTED** [review]

File: packages/language-service/src/typescript_host.ts (hunk 1 above)
Comment (ayazhafiz) [reply packages/language-service/src/typescript_host.ts]: Ah, my thinking was inverted. thanks!

**Review by rbuzzibarrios — APPROVED** [review]

**Review by kyliau — APPROVED** [review]
